src/uct/ib/mlx5/ib_mlx5.c

```diff
@@ -348,8 +401,106 @@ ucs_status_t uct_ib_mlx5_get_compact_av(uct_ib_iface_t *iface, int *compact_av)
 }
 #endif
 
-void uct_ib_mlx5_check_completion(uct_ib_iface_t *iface, uct_ib_mlx5_cq_t *cq,
-                                  struct mlx5_cqe64 *cqe)
+static void
+uct_ib_mlx5_iface_cqe_unzip_fill_unique(struct mlx5_cqe64 *cqe,
+                                        uct_ib_mlx5_mini_cqe8_t *mini_cqe,
+                                        uct_ib_mlx5_cq_unzip_t *cq_unzip)
+{
+    const uint32_t net_qpn_mask = htonl(UCS_MASK(UCT_IB_QPN_ORDER));
+    uint16_t new_wqe_opcode;
+
+    /* TODO: Think about the checksum field unzipping
+     * The checksum field does not copied from src dst
+     * because of absence of the mlx5_cqe64::check_sum field on some systems.
+     */
+    cqe->byte_cnt = mini_cqe->byte_cnt;
+    ucs_assert(!(cqe->op_own & UCT_IB_MLX5_CQE_OP_OWN_ERR_MASK));
+    if ((cqe->op_own >> 4) == MLX5_CQE_REQ) {
+        cqe->wqe_counter  = mini_cqe->wqe_counter;
+        cqe->sop_drop_qpn = (cqe->sop_drop_qpn & net_qpn_mask) |
+                            htonl(mini_cqe->s_wqe_opcode << UCT_IB_QPN_ORDER);
+    } else if (cq_unzip->block_size != cq_unzip->left) {
+        /* The title_cqe opcode shouldn't be updated */
+        new_wqe_opcode = ntohs(cqe->wqe_counter) + 1;
+        cqe->wqe_counter = htons(new_wqe_opcode);
+    }
```

Comment:
looks weird. 
1. why +1?
2. what if (cq_unzip->block_size == cq_unzip->left) - we don't need to fill wqe counter?
3. why opcode mixed with counter?
4. don't we need to fill sop_drop_qpn?